### PR TITLE
TST: Fixed regex to hold gcc version

### LINKED ISSUE #2185: all bot test fail on Ubuntu 22.04 because of differing python version string
for example:
```
[  213s] AssertionError: Regex didn't match: 'ShadowserverJSONParserBot initialized with id test-bot and intelmq [0-9a-z.]* and python [0-9a-z.]{5,8}\\+? \\([a-zA-Z0-9,:. ]+\\)( \\[GCC\\])? as process [0-9]+\\.' not found in 'ShadowserverJSONParserBot initialized with id test-bot and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.'
```
applies to all bot tests of course
`( \\[GCC\\])?` needs to be extend to also hold the gcc version (or a non-greedy catch-all)

### PR DESCRIPTION
**Pull request by waldbauer-certat**
Fixes #2185

### CODE REVIEW

File: intelmq/lib/test.py [hunk 1]
```diff
@@ -348,8 +348,11 @@ def run_bot(self, iterations: int = 1, error_on_pipeline: bool = False,
                 self.assertIn('raw', event)
 
         """ Test if bot log messages are correctly formatted. """
-        self.assertLoglineMatches(0, "{} initialized with id {} and intelmq [0-9a-z.]* and python"
-                                     r" [0-9a-z.]{{5,8}}\+? \(.+?\)( \[GCC.*?\])?"
+        # regex credits for semver: https://ihateregex.io/expr/semver/
+        self.assertLoglineMatches(0, "{} initialized with id {} and"
+                                     r" intelmq [0-9a-z.]*"
+                                     r" and python [0-9a-z.]*"
+                                     r" \(.+?\)(?:\ \[GCC.*?\])?"
```
Comment (sebix) [inline intelmq/lib/test.py]:
```suggestion
        self.assertLoglineMatches(0, "{} initialized with id {} and"
                                     r" intelmq [0-9a-z.]*"
                                     r" and python [0-9a-z.]*"
                                     r" \(.+?\)( \[GCC.*?\])?"
```
Your regex (or that of ihateregex) did not work with the string `ShadowserverJSONParserBot initialized with id test-bot and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.`

**Review by sebix — CHANGES_REQUESTED** [review]

File: intelmq/lib/test.py (hunk 1 above)
Comment (waldbauer-certat) [reply intelmq/lib/test.py]: Ah, its not ihateregex, missed to remove credits. Sorry

**Comment by sebix** [comment]
Did you test the regex with the string `ShadowserverJSONParserBot initialized with id test-bot and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.`?
It would be easier to test and check with a test case for the regular expression.

**Comment by wagner-intevation** [comment]
@waldbauer-certat I added tests. Please merge if it looks sane to you.

**Review by sebix — APPROVED** [review]

**Pull request merged**